### ISSUE TITLE
door opener buttons on 3rd floor stopped working

### ISSUE THREAD

**Issue by tingox** [bitraf/bitraf-iot#13]
Today, office users reported that the open-door buttons on the 3rd floor didn't work. Opening the door with door.bitraf.no worked. I checked and found that:
- both buttons (workshop and office3rdfloor) was online and sent mqtt messages, also when the button was pressed.
- the service that tells the doors to open was running
- opening the doors via door.bitraf.no was working
info about the service
```
root@bomba:~# systemctl status bitraf-iot-opendoor3floor.service
* bitraf-iot-opendoor3floor.service
   Loaded: loaded (/etc/systemd/system/bitraf-iot-opendoor3floor.service; enable
   Active: active (running) since Thu 2017-11-09 15:35:52 CET; 1 weeks 0 days ag
 Main PID: 23348 (msgflo-python)
    Tasks: 1 (limit: 4915)
   CGroup: /system.slice/bitraf-iot-opendoor3floor.service
           `-23348 /usr/bin/python /usr/local/bin/msgflo-python components/OpenD

Warning: Journal has been rotated since unit was started. Log output is incomplete

```
and
```
root@bomba:~# journalctl _SYSTEMD_UNIT=bitraf-iot-opendoor3floor.service
-- No entries --

```
I restarted the service
`root@bomba:~# systemctl restart bitraf-iot-opendoor3floor.service
`

and the buttons work again.

**Comment by tingox** [comment]
Normal status for the service
```
root@bomba:~# systemctl status bitraf-iot-opendoor3floor.service
* bitraf-iot-opendoor3floor.service
   Loaded: loaded (/etc/systemd/system/bitraf-iot-opendoor3floor.service; enabled; vendor preset: enabled)
   Active: active (running) since Fri 2017-11-17 10:56:01 CET; 13min ago
 Main PID: 16647 (msgflo-python)
    Tasks: 1 (limit: 4915)
   CGroup: /system.slice/bitraf-iot-opendoor3floor.service
           `-16647 /usr/bin/python /usr/local/bin/msgflo-python components/OpenDoor3floor.py opendoor3floor

Nov 17 10:56:01 bomba systemd[1]: Started bitraf-iot-opendoor3floor.service.
Nov 17 10:56:01 bomba msgflo-python[16647]: components OpenDoor3floor
Nov 17 10:56:01 bomba msgflo-python[16647]: opendoor3floor(OpenDoor3floor) running on mqtt://mqtt.bitraf.no

```
and journalctl
```
root@bomba:~# journalctl _SYSTEMD_UNIT=bitraf-iot-opendoor3floor.service
-- Logs begin at Fri 2017-11-17 10:33:08 CET, end at Fri 2017-11-17 11:09:56 CET. --
Nov 17 10:56:01 bomba msgflo-python[16647]: components OpenDoor3floor
Nov 17 10:56:01 bomba msgflo-python[16647]: opendoor3floor(OpenDoor3floor) running on mqtt://mqtt.bitraf.no

```

**Comment by tingox** [comment]
Speculation: could it be that this component can't handle a restart of the MQTT service when it is running?
Not very likely, because the other components are still working, and the code is very similar.

**Comment by jonnor** [comment]
Yes, this is a bug in msgflo-python https://github.com/msgflo/msgflo-python/issues/32

**Comment by jonnor** [comment]
This also affects door opener on second floor / lab, incl the status light.
Easiest way to reproduce: restart the MQTT broker

**Comment by jonnor** [comment]
Reconnect issue should be fixed in msgflo-python 0.2.3 - will test tomorrow

**Comment by tingox** [comment]
The fix was tested and found working some time ago.
